### PR TITLE
Alert when upload-components field is empty #135

### PR DESCRIPTION
**Pull request by johnmhoran**
* The upload process will not start if the value of any form field
    is “”.

Signed-off-by: John M. Horan <johnmhoran@gmail.com>

### CODE REVIEW

File: assets/js/renderer.js [hunk 1]
```diff
@@ -646,32 +646,42 @@ $(document).ready(function () {
...
-                        owner: component.owner,
-                        license_expression: component.license_expression,
-                        copyright: component.copyright,
-                        homepage_url: component.homepage_url,
-                        primary_language: component.programming_language,
-                        reference_notes: component.notes,
-                        product: productNameVersion
-                    }
-                });
+        let fieldProductName = $("#product-name").val();
```
Comment (jdaguil) [inline assets/js/renderer.js]: Avoid repeating code by not redefining these variables. You can use the variables that are already defined below. Also, you can change the references from `let` to `const`.

File: assets/js/renderer.js [hunk 2]
```diff
@@ -646,32 +646,42 @@ $(document).ready(function () {
...
+        let fieldProductVersion = $("#product-version").val();
+        let fieldApiUrl = $("#apiURLDejaCode").val();
+        let fieldApiKey = $("#apiKey").val();
+        // Test whether any form field is empty
+        if ((fieldProductName == "") || (fieldProductVersion == "") || (fieldApiUrl == "") || (fieldApiKey == "")) {
```
Comment (jdaguil) [inline assets/js/renderer.js]: Could you change `==` to `===`?

**Comment by johnmhoran** [comment]
Thanks @jdaguil .  I'll turn to these cleanups tomorrow.  

Can you suggest one or two good validation libraries I should look into?

**Comment by jdaguil** [comment]
@johnmhoran possible libraries: 
https://jqueryvalidation.org/
http://1000hz.github.io/bootstrap-validator/
http://parsleyjs.org/doc/index.html

Changes LGTM!

**Comment by johnmhoran** [comment]
Thanks @jdaguil -- I'll give your suggestions a close look.  Much appreciated!

**Pull request merged**